masonry/src/widget/window_handle.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright 2024 the Xilem Authors and the Druid Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! A titlebar area widget.
+
+use accesskit::{NodeBuilder, Role};
+use dpi::LogicalPosition;
+use smallvec::{smallvec, SmallVec};
+use tracing::{trace_span, Span};
+use vello::kurbo::Point;
+use vello::Scene;
+
+use crate::event::PointerButton;
+use crate::widget::{WidgetMut, WidgetPod};
+use crate::{
+    AccessCtx, AccessEvent, BoxConstraints, EventCtx, LayoutCtx, LifeCycleCtx, PaintCtx,
+    PointerEvent, Size, StatusChange, TextEvent, Widget, WidgetId,
+};
+
+// TODO: Maybe make this configurable somehow
+const DRAG_THRESHOLD: f64 = 8.0;
+
+/// A titlebar area widget.
+///
+/// An area that can be dragged to move the window.
```

Comment:
This needs some more documentation about how it's expected to be used.

Reply:
Done, I hope it's clear enough.

Reply:
Just discovered that floem has something similar, if that can help: https://github.com/lapce/floem/blob/main/src/views/drag_window_area.rs